README.md

```diff
@@ -1,29 +1,111 @@
-:duck: Malduck
-=========
+# :duck: Malduck
 
 Malduck is your ducky companion in malware analysis journeys. It is mostly based on [Roach](https://github.com/hatching/roach) project, which derives many concepts from [mlib](https://github.com/mak/mlib) 
 library created by [Maciej Kotowicz](https://lokalhost.pl). The purpose of fork was to make Roach independent from [Cuckoo Sandbox](https://cuckoosandbox.org/) project, but still supporting its internal `procmem` format.
 
 Malduck provides many improvements resulting from CERT.pl codebase, making scripts written for malware analysis purposes much shorter and more powerful. 
 
-Improvements
-============
+## Features
 
-* Support for (non)memory-mapped PE images without header fix-up.
-* Searching for wildcarded byte sequences
-* Support for x64 disassembly
-* Fixed-precision integer types
-* Many improvements in ProcessMemory
+- **Cryptography** (AES, Blowfish, Camelie, ChaCha20, Serpent and many others)
+- **Compression algorithms** (aplib, gzip, inznt1 (RtlDecompressBuffer)
+- **Memory model objects** (work on memory dumps, PE/ELF, raw files and IDA dumps with the same code)
+- **Extraction engine** (modular extraction framework for config extraction from files/dumps)
+- Fixed integer types (like Uint64) and bitwise utilities
+- String operations (chunks, padding, packing/unpacking etc)
+- Hashing algorithms (CRC32, MD5, SHA1, SHA256)
 
-Usage
-==========
+## Usage examples
 
-Installing may be performed by running
+#### AES
+
+```python
+from malduck import aes
+
+key = b'A'*16
+iv = b'B'*16
+plaintext = b'data'*16
+ciphertext = aes.cbc.encrypt(key, iv, plaintext)
+```
+
+### Serpent
+
+```python
+from malduck import serpent
+
+key = b'a'*16
+iv = b'b'*16
+plaintext = b'data'*16
+ciphertext = serpent.cbc.encrypt(key, plaintext, iv=iv)
+```
+
+### APLib decompression
+
+```python
+from malduck import aplib
+
+# Headerless compressed buffer
+aplib(b'T\x00he quick\xecb\x0erown\xcef\xaex\x80jumps\xed\xe4veur`t?lazy\xead\xfeg\xc0\x00')
+```
+
+### Fixed integer types
+
+```python
+from malduck import DWORD
+
+def sdbm_hash(name: bytes):
+    hh = 0
+    for c in name:
+        hh = DWORD(c) + (hh << 6) + (hh << 16) - hh
```

Comment:
I think a small comment here that explains how the DWORD type is propagated to the rest of the variables could be useful